SECURITY-INSIGHTS.yml

```diff
@@ -0,0 +1,38 @@
+header:
+  schema-version: '1.0.0'
+  expiration-date: '2025-01-25T01:00:00.000Z'
+  last-updated: '2024-01-25'
+  last-reviewed: '2024-01-25'
+  project-url: https://github.com/cilium/cilium
+  license: https://github.com/cilium/cilium/blob/main/LICENSE
+project-lifecycle:
+  status: active
+  bug-fixes-only: false
+  core-maintainers:
+    - https://github.com/cilium/cilium/blob/main/MAINTAINERS.md
+  roadmap: https://docs.cilium.io/en/latest/community/roadmap
+contribution-policy:
+  accepts-pull-requests: true
+  accepts-automated-pull-requests: true
+dependencies:
+  third-party-packages: true
+  dependencies-lists:
+    - https://github.com/cilium/cilium/blob/main/go.mod
+distribution-points:
+  - https://github.com/cilium/cilium
+documentation:
+  - https://docs.cilium.io/en/stable/
+security-contacts:
+  - type: email
+    value: security@cilium.io
+security-testing:
+- tool-type: sast
+  tool-name: CodeQL
+  tool-version: '2.22.12'
+  integration:
+    ad-hoc: false
+    ci: true
+    before-release: true
```

Comment:
I'm not sure if this list is comprehensive. For instance there is an oss-fuzz project, and there are other things. Additionally, I believe that CodeQL is currently disabled due to lack of interest from community to maintain it.

Reply:
I see, if you have something more precise, please share : https://github.com/ossf/security-insights-spec/blob/v1.0.0/specification.md#security-testing

Reply:
I've pulled in @ferozsalam for review, maybe he has a better sense for some of these.

Reply:
Thanks for the link @mmorel-35. As Joe has mentioned, we're not using CodeQL at the moment, so that should be removed.

Looking at the examples in the link, we would probably also want to add the following:
- [Renovate](https://github.com/cilium/cilium/issues/22100)
- oss-fuzz
- [Container vulnerability scanning with grype](https://github.com/cilium/cilium/actions/workflows/container-scan.yaml)

I notice that the `tool-version` is a required field for this file according to the spec. We are continuously updating the versions of all the tools that we use, so I'm not sure this would be useful to define again here for our Grype test, for example. For Renovate, which acts as an SCA tool for us that's equivalent to Dependabot, I can't see a way of identifying a version. Do you have any thoughts on how we can represent these cases without the information in the file becoming outdated almost immediately?

On the semantics of 'before release' – all of the tooling described above runs continuously, but none of them gate a release. Would `before-release` be true or false in these cases?

Reply:
I removed the CodeQL.

Concerning the matter of tool-version, I have seen 'latest' use in some projects so I believe that can be used here too.

About before-release I would say it’s a true. As the description seems wide all use of those tool happens before release. As they are not used during runtime for example.  I’m just interpreting here as I’m not the one who wrote the specification.